### PR TITLE
[chore] Ensure tests fail

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:** 
Fixes the issue uncovered here https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17037

### CODE REVIEW

**Comment by MovieStoreGuy** [comment]
It is expected the tests should fail now once this is done.

**Comment by mx-psi** [comment]
The change looks good to me and it looks like the tests are indeed failing. I can't bypass the checks so we need to solve the unit tests issues on this PR too

File: exporter/datadogexporter/traces_exporter_test.go [hunk 1]
```diff
@@ -290,6 +290,8 @@ func TestNewTracesExporter(t *testing.T) {
 }
 
 func TestPushTraceData(t *testing.T) {
+	t.Skip("Skipping as a result of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17037")
```
Comment (MovieStoreGuy) [inline exporter/datadogexporter/traces_exporter_test.go:293]: I've marked this as Skip for now @mx-psi , and if you could revert once your change has gone through?

**Review by MovieStoreGuy — COMMENTED** [review]
So far I have tried to fix things where needed and tidied things up while I was there.

File: exporter/datadogexporter/traces_exporter_test.go (hunk 1 above)
Comment (mx-psi) [reply exporter/datadogexporter/traces_exporter_test.go:293]: Sounds good to me, I opened #17045 to track fixing this and assigned to me

File: processor/spanmetricsprocessor/processor.go [hunk 2]
```diff
@@ -32,6 +32,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/atomic"
```
Comment (mx-psi) [inline processor/spanmetricsprocessor/processor.go:35]: I would rather revert the offending PR than do this change, do you know which one failed?

File: receiver/prometheusreceiver/internal/staleness_end_to_end_test.go [hunk 3]
```diff
@@ -140,17 +140,18 @@ service:
       exporters: [prometheusremotewrite]`, serverURL.Host, prweServer.URL)
 
 	confFile, err := os.CreateTemp(os.TempDir(), "conf-")
-	require.Nil(t, err)
```
Comment (mx-psi) [inline receiver/prometheusreceiver/internal/staleness_end_to_end_test.go:143]: Are these changes necessary to fix CI?

File: processor/spanmetricsprocessor/processor_test.go [hunk 4]
```diff
@@ -1026,6 +1029,7 @@ func TestProcessorUpdateExemplars(t *testing.T) {
 }
 
 func TestConsumeTracesEvictedCacheKey(t *testing.T) {
+	t.Skip("Skipping as a result of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17037")
```
Comment (mx-psi) [inline processor/spanmetricsprocessor/processor_test.go:1032]: Can you create an issue for this?

File: receiver/prometheusreceiver/internal/staleness_end_to_end_test.go (hunk 3 above)
Comment (MovieStoreGuy) [reply receiver/prometheusreceiver/internal/staleness_end_to_end_test.go:143]: I couldn't help myself... but no

File: Makefile.Common [hunk 5]
```diff
@@ -1,3 +1,5 @@
+SHELL=/bin/bash -o pipefail
```
Comment (dmitryax) [inline Makefile.Common:1]: Does it mean we route all `stdout` to `stderr`?

Comment (MovieStoreGuy) [reply Makefile.Common:1]: Nah, so this just forces that bash keeps track of the entire pipe command instead of the last entry in the pipe command group.

**Comment by MovieStoreGuy** [comment]
Reverted my fixes to see what remains

**Comment by dmitryax** [comment]
@MovieStoreGuy did you rebase on the latest main? I think I see errors that should be fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16601

File: Makefile.Common (hunk 5 above)
Comment (dmitryax) [reply Makefile.Common:1]: Can you please add a comment on what it does and why it's required for the test?

**Comment by MovieStoreGuy** [comment]
I thought I had but I had missed some, going again

File: Makefile.Common (hunk 5 above)
Comment (MovieStoreGuy) [reply Makefile.Common:5]: Let me know if that comment makes sense.

Comment (dmitryax) [reply Makefile.Common:5]: That's great. Thank you!

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@MovieStoreGuy I think all tests should be resolved now. Please rebase

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**